pallets/liquidity-pools/src/message.rs

```diff
@@ -367,616 +412,70 @@ where
 	///
 	/// Directionality: Centrifuge -> EVM Domain.
 	DisallowInvestmentCurrency {
-		pool_id: PoolId,
+		pool_id: u64,
 		currency: u128,
 	},
 }
 
-impl<
-		Domain: Codec,
-		PoolId: Encode + Decode,
-		TrancheId: Encode + Decode,
-		Balance: Encode + Decode,
-		Ratio: Encode + Decode,
-	> Message<Domain, PoolId, TrancheId, Balance, Ratio>
-{
-	/// The call type that identifies a specific Message variant. This value is
-	/// used to encode/decode a Message to/from a bytearray, whereas the head of
-	/// the bytearray is the call type, followed by each message's param values.
-	///
-	/// NOTE: Each message must immutably  map to the same u8. Messages are
-	/// decoded in other domains and MUST follow the defined standard.
-	fn call_type(&self) -> u8 {
```

Comment:
Just to be sure: The call type is now inherited from the order of the enum variants, right? So when I have a nested enum `Y` inside an LP Message `X`, `Y` will have its own inner call type such that `X` has an outer and an inner one.

Reply:
> The call type is now inherited from the order of the enum variants, right?

Yes

> So when I have a nested enum Y inside an LP Message X, Y will have its own inner call type such that X has an outer and an inner one.

And yes too

Reply:
Probably we should have a test checking that each msg has the expected index in the spec, to avoid errors swapping variants in the future.

Reply:
Yeah I was thinking of exactly that. It is more or less tested in the message serialization cycle UTs in `Message.rs` but these are too opaque to guarantee call type safety. No need for you to do it as part of this PR because I am changing almost all indices anyway. Will do it as part of #1892 then.

Reply:
> It is more or less tested in the message serialization cycle UTs in Message.rs but these are too opaque to guarantee call type safety

fully agree!

Reply:
To avoid boilerplate in testing, maybe each test construction can go to a function: i.e. `increase_invest_order_msg()`, and later:

```rust
`assert_eq!(msg_increase_invest_order().serialize()[0], 5)`
```

or similar.

Also, not sure if the it's more manageable to have the message as:

```rust
enum Message {
    AddPool(AddPoolMessage),
    AllowInvestCurrency(AllowInvestCurrencyMessage),
    ..
},
```

It seems like "more" code at first, but later, you can reuse each variant for different purposes. I'm just thinking out loud.

Reply:
> Also, not sure if the it's more manageable to have the message as:
> 
> ```rust
> enum Message {
>     AddPool(AddPoolMessage),
>     AllowInvestCurrency(AllowInvestCurrencyMessage),
>     ..
> },
> ```

Not sure if I see the win for doing it this way right now.